src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -924,35 +924,14 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
         }
     };
 
-    const auto reorder_input_concat = [&p, &rf](typed_program_node<concatenation>& concat_node) {
```

Comment:
Have you checked perf impact? Looks like removal of this part may lead to disabled concat optimization and explicit kernel execution which will drop the performance.

Reply:
> Have you checked perf impact? Looks like removal of this part may lead to disabled concat optimization and explicit kernel execution which will drop the performance.

The code was introduced by PR #25891 and it fixed the issue that reorder was not added in a rare condition. The rare condition was: the selected format in selected_preferred_format was bfyx but fsv32 in reorder_input, and the reorder should be added in this case as you commented by the optimization, but it was not added.
So the PR #25891 fixed the issue by adding reorder in such a rare case. but it seemed to add reorder in a rare case. So this PR changed the approach as shape agnostic concat ref kernel handled the rare case. And as a result, no perf impact was expected. Let me know if anything was missed.

Reply:
Well, it's good that perf impact is not expected, but have you verified that? 😄

Reply:
Thanks for you comments! Just checked it using the issued model, and it showed 75.41 FPS : 74.70 FPS (PR : Non-PR).

Reply:
From dGPU static shape model daily test scope, there was no performance impact